Cleaned up Go code directory structure

p-t-u opened the pull request:
- **build-ocb now also works under Linux**
- **Moved cmd/collector to observecol**
- **Renamed/moved directory cmd to internal**
- **Canonicalized the naming of our Go modules**

### Description

OB-37278: Clean up Go code directory structure

obs-gh-alexlew commented:
no goreleaser changes required? changing the package name didn't affect the executable name?

obs-gh-alexlew commented on observecol/otelcollector.go:
```diff
@@ -24,16 +22,16 @@ func makeMapProvidersMap(providers ...confmap.Provider) map[string]confmap.Provi
+func GenerateCollectorSettings(URIs []string) *otelcol.CollectorSettings {
 	buildInfo := component.BuildInfo{
-		Command:     "observe-agent",
+		Command:     "github.com/observeinc/observe-agent",
```
should this have been changed to the full package path as well?

p-t-u replied:
Thanks for spotting. This is a copy-paste artefact. I'll fix and look for any other instances.

obs-gh-alexlew commented on internal/root/root.go:
```diff
@@ -16,7 +16,7 @@ var CfgFile string
 // RootCmd represents the base command when called without any subcommands
 var RootCmd = &cobra.Command{
-	Use:   "observe-agent",
+	Use:   "github.com/observeinc/observe-agent",
```
same here, this update doesn't seem necessary. `observe-agent` is still the executable right?

p-t-u replied:
Yes. Ditto.

obs-gh-alexlew commented on internal/root/root.go:
```diff
@@ -64,7 +64,7 @@ func InitConfig() {
 		viper.AddConfigPath(config.GetDefaultAgentPath())
 		viper.SetConfigType("yaml")
-		viper.SetConfigName("observe-agent")
+		viper.SetConfigName("github.com/observeinc/observe-agent")
```
ditto above

obs-gh-alexlew commented on internal/config/confighandler.go:
```diff
@@ -103,7 +102,7 @@ func GetDefaultAgentPath() string {
 	switch currOS := runtime.GOOS; currOS {
 	case "darwin":
-		return filepath.Join(os.Getenv("HOME"), "Library", "Application Support", "Observe", "observe-agent", "filestorage")
+		return filepath.Join(os.Getenv("HOME"), "Library", "Application Support", "Observe", "github.com/observeinc/observe-agent", "filestorage")
```
dont think this needs the github in the dirname

p-t-u commented:
> no goreleaser changes required? changing the package name didn't affect the executable name?

I don't think so. `goreleaser` is still happy. Here's some relevant output from the snapshot build:

```
    • added new artifact                             name=observe-agent type=Binary path=dist/default_linux_386/observe-agent                                                                            
    • added new artifact                             name=observe-agent type=Binary path=dist/default_linux_arm64/observe-agent    
    • added new artifact                             name=observe-agent type=Binary path=dist/default_darwin_amd64_v1/observe-agent                                                                      
    • added new artifact                             name=observe-agent.exe type=Binary path=dist/windows_build_windows_386/observe-agent.exe                                                            
    • added new artifact                             name=observe-agent type=Binary path=dist/default_linux_amd64_v1/observe-agent                                                                       
    • added new artifact                             name=observe-agent.exe type=Binary path=dist/windows_build_windows_amd64_v1/observe-agent.exe                                                       
    • added new artifact                             name=observe-agent type=Binary path=dist/default_darwin_arm64/observe-agent
```

The final segment of the package name is also still "observe-agent", so no change needed there. And the "github.com..." stuff is of course never part of the executable name.

goreleaser eventually fails with these kind of errors:
```
#11 [directories 2/2] RUN mkdir -p /var/lib/observe-agent/filestorage                                                                                                                                    
#11 0.490 exec /bin/sh: exec format error                                                                                                                                                                
#11 ERROR: process "/bin/sh -c mkdir -p /var/lib/observe-agent/filestorage" did not complete successfully: exit code: 1
```
but I read that's normal for local builds outside of GitHub CI, and it happens after the executables are built successfully.
